packages/core/test/reducers/core.test.ts

```diff
@@ -2166,3 +2166,297 @@ test('core reducer - REQUIRED rule updates schema required fields when the condi
     'taxId should not be required when type is person'
   );
 });
+
+test('core reducer - SHOW rule with preserveValueOnHide preserves field values when hidden', (t) => {
+  // Create a schema with fields that will be conditionally shown/hidden
+  const schema = {
+    type: 'object',
+    properties: {
+      type: { type: 'string', enum: ['person', 'organization'] },
+      personalInfo: {
+        type: 'object',
+        properties: {
+          firstName: { type: 'string' },
+          lastName: { type: 'string' },
+        },
+      },
+      organizationInfo: {
+        type: 'object',
+        properties: {
+          companyName: { type: 'string' },
+          registrationNumber: { type: 'string' },
+        },
```

Comment:
Seems like we should only need one property for this test? Can we get rid of the extras ?